apps/api/src/utils/aliases.ts

```diff
@@ -0,0 +1,197 @@
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import {
+	type ApplicationContentUpdates,
+	type ApplicationSignatureUpdate,
+	type CollaboratorRecord,
+	type JoinedApplicationRecord,
+} from '@/service/types.js';
+import {
+	type ApplicationContentsResponse,
+	type ApplicationResponseData,
+	type CollaboratorsResponse,
+	type SignatureDTO,
+} from '@pcgl-daco/data-model/src/types.js';
+import { type UpdateEditApplicationRequest } from '@pcgl-daco/validation';
+import lodash from 'lodash';
+
+const { camelCase, snakeCase } = lodash;
+
+type inputKey = string & keyof Record<string, any>;
+
+/** Used in filter functions for alias utilities below
+ * @param key Current key to validate
+ * @param omittedKeys List of keys to remove from output
+ */
+const filterOmittedKeys = (omittedKeys: string[]) => (key: string) => !omittedKeys.includes(key);
+
+/**
+ * Helper function to convert Postgres snake_case to FE camelCase
+ * Generics allow usage w/ multiple input/output combinations
+ * @param data Original Database record with snake case keys to convert
+ * @param omittedKeys List of keys to remove from output for cases where a partial record is returned
+ * @returns ResponseRecord - Generic type representing new record with updated camelCase keys
+ */
+export const convertToCamelCase = <
+	SnakeCaseRecord extends Record<string, any>,
+	CamelCaseRecord extends Record<string, any>,
+>(
+	data: SnakeCaseRecord,
+	omittedKeys: string[] = [],
+): CamelCaseRecord => {
+	const allKeys: inputKey[] = Object.keys(data);
+
+	const filteredKeys = allKeys.filter(filterOmittedKeys(omittedKeys));
+
+	const responseData = filteredKeys.reduce((acc, key) => {
+		const aliasedKey = camelCase(key);
+		const value = data[key];
+		const accumulator = { ...acc, [aliasedKey]: value };
+		return accumulator;
+	}, {}) as CamelCaseRecord;
+
+	return responseData;
+};
+
+/**
+ * Helper function to convert FE camelCase to Postgres snake_case
+ * Generics allow usage w/ multiple input/output combinations
+ * @param data Original Request record with camelCase keys to convert
+ * @param omittedKeys List of keys to remove from output for cases where a partial record is returned
+ * @returns DatabaseRecord - Generic type representing new record with updated snake_case keys
```

Comment:
This doesn't return a generic type. This returns the object you describe here.

Reply:
This comment (and function) have been removed